### PR TITLE
Pa 8

### PR DESCRIPTION
**Pull request by avb49**
Added logic to User Capabilities - changes to the preferences screen are now saved for a given user to the database and upon creation of the fragment (preferences) the database is consulted for the values of the user's properties. Works for both users that do have preferences already in the database and also users that do not (in this case, an entry in the database is created for them in the user_properties table and default values set).

### CODE REVIEW

File: app/src/main/java/uk/ac/bath/petmatch/UserCapabilitiesActivity.java [hunk 1]
```diff
@@ -1,27 +1,69 @@
...
 
-                return;
-            }
-            getFragmentManager().beginTransaction().add(R.id.fragment_container, new UserCapabilitiesFragment()).commit();
+            UserProperties newUserCapabilities = new UserProperties(false, false,
+                    false, false, false, currentUser);
+            userPropertiesDao.create(newUserCapabilities);
         }
+        userCapabilitiesFragment = new UserCapabilitiesFragment();
+        getFragmentManager().beginTransaction().add(R.id.fragment_container, userCapabilitiesFragment).commit();
```
Comment (novotnej) [inline app/src/main/java/uk/ac/bath/petmatch/UserCapabilitiesActivity.java]: getFragmentManager() is deprecated, is there another way of doing this?

File: app/src/main/java/uk/ac/bath/petmatch/Utils/UserCapabilitiesFragment.java [hunk 2]
```diff
@@ -1,17 +1,45 @@
 package uk.ac.bath.petmatch.Utils;
```
Comment (novotnej) [inline app/src/main/java/uk/ac/bath/petmatch/Utils/UserCapabilitiesFragment.java]: Please remove fragment from Utils package, either to the main package or create Fragments package

File: app/src/main/java/uk/ac/bath/petmatch/MainActivity.java [hunk 3]
```diff
@@ -186,12 +204,15 @@ public boolean onNavigationItemSelected(MenuItem item) {
...
-            // handles user capabilities
-            Intent startUserCapabilitiesIntent = new Intent(getApplicationContext(),
-                    UserCapabilitiesActivity.class);
-            startActivity(startUserCapabilitiesIntent);
+            if(loginService.isUserLoggedIn()) {
+                Intent startUserCapabilitiesIntent = new Intent(getApplicationContext(),
+                        UserCapabilitiesActivity.class);
+                startActivity(startUserCapabilitiesIntent);
+            } else {
+                displayNoLoginErrorAlert();
```
Comment (novotnej) [inline app/src/main/java/uk/ac/bath/petmatch/MainActivity.java]: I really like the message, but also put it inside the UserCapabilitiesActivity, if a user is accidentally directed there and is not logged in

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 4]
```diff
@@ -17,9 +17,9 @@
             android:icon="@drawable/ic_menu_slideshow"
             android:title="Shelter Profile" />
         <item
-            android:id="@+id/nav_settings"
+            android:id="@+id/nav_user_capabilities"
```
Comment (novotnej) [inline app/src/main/res/menu/activity_main_drawer.xml:21]: in the mainactivity, perhaps in the setUpLoginView or whatever I called it, or after login, only show this when user is logged in (hidden by default). Then hide it again when user logs out. 

Similar logic to the login/logout buttons

**Review by novotnej — CHANGES_REQUESTED** [review]
Really impressed you got the fragment and activity working together. 

Some details that need fixing but otherwise good work

**Comment by avb49** [comment]
> Really impressed you got the fragment and activity working together.
> 
> Some details that need fixing but otherwise good work

Cheers boss

**Review by novotnej — APPROVED** [review]

**Comment by novotnej** [comment]
You still don't have the if user is logged in check in the activity, but nvm that for now, we need to move on.. :D

**Pull request merged**